viite-UI/src/view/RoadAddressChangesBrowserWindow.js

```diff
@@ -0,0 +1,331 @@
+            '<div style="display: flex; flex-direction: column; padding: 5px; white-space: nowrap">' +
+                '<div style="display: flex; flex-direction: row"><input type="radio" name="roadAddrChangesBrowserForm" value="ProjectAcceptedDate" checked="checked"><label style="margin-left: 5px; margin-top: 2.5px">Projektin hyväksymispvm</label></div>' +
+                '<div style="display: flex; flex-direction: row"><input type="radio" name="roadAddrChangesBrowserForm" value="RoadAddressStartDate"><label style="margin-left: 5px; margin-top: 2.5px">Muutoksen voimaantulopvm</label></div>' +
+            '</div>' +
+            '<button class="btn btn-primary btn-fetch-road-address-changes"> Hae </button>' +
+            '<button id="exportAsExcelFile" class="download-excel btn" disabled>Lataa Excelinä <i class="fas fa-file-excel"></i></button>' +
+            '</form>'
+        );
+
+        function showResults(results) {
+            const arr = [];
+            let arrPointer = -1;
+            arr[++arrPointer] = `<table id="roadAddressChangesBrowserTable" class="road-address-browser-window-results-table">
+                                    <thead>
+                                        <tr>
+                                            <th>Voimaantulopvm</th>
+                                            <th>Ely</th>
+                                            <th>Tie</th>
+                                            <th>Ajr</th>
+                                            <th>Aosa</th>
+                                            <th>Aet</th>
+                                            <th>Losa</th>
+                                            <th>Let</th>
+                                            <th>Pituus</th>
+                                            <th>Hall. luokka</th>
+                                            <th>Muutos</th>
+                                            <th>u_Ely</th>
+                                            <th>u_Tie</th>
+                                            <th>u_Ajr</th>
+                                            <th>u_Aosa</th>
+                                            <th>u_Aet</th>
+                                            <th>u_Losa</th>
+                                            <th>u_Let</th>
+                                            <th>u_Pituus</th>
+                                            <th>u_Hall. luokka</th>
+                                            <th>Käännetty</th>
+                                            <th>Tien nimi</th>
+                                            <th>Projektin Nimi</th>
+                                            <th>Projektin hyväksymispvm</th>
+                                        </tr>
+                                    </thead>
+                                    <tbody>`;
+
+            for (let i = 0, len = results.length; i < len; i++) {
```

Comment:
Mihin indexsiä tarvitaan ? Eikös tän pysty tekemään yksinkertaistetulla for loop
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach

Reply:
Pystyy ja niin se oli tehtykin mutta tähän päädyttiin performance syistä.

https://blog.scottlogic.com/2010/10/15/javascript-array-performance.html

Reply:
Jos performassi syy, kannattaa lisätä kommenti siitä koska seuraava lukia ei voi tietää miksi päätettiin käyttää tätä eikä foreach,
 
tietty on ton vois kyl tehdä myös for (item in list) {} muodossa myös.

Reply:
Tuliko mitattua kuis paljon tää nyt parantaa nopeutta. Pieni arvelus että näissä otoksissa ero on aika olematon.

Reply:
Tuli testattua ja indexin käyttö oli nopein ratkaisu

Reply:
Suurilla data määrillä käyttäjä joutuu odottamaan tällä hetkellä pari sekuntia joten tämä kohta halutiin niin nopeaksi kuin mahdollista

Reply:
for/in loop taitaa olla yksi hitaimmista tavoista loopata javascriptissa 🤔

Reply:
https://blog.bitsrc.io/measuring-performance-of-different-javascript-loop-types-c0e9b1d193ed

Reply:
Joo varmaan voit soittaa V8 kehittäjille miten ne toteutti ton for  loop määrityksen sinne, mut joo koodiin kommetti että performansi kriittinen kohta ja jos muutat muista mitata muutos.